backends/cadence/hifi/operators/op_add.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/kernels/portable/cpu/scalar_utils.h>
+#include <executorch/kernels/portable/cpu/util/broadcast_util.h>
+#include <executorch/kernels/portable/cpu/util/functional_util.h>
+#include <executorch/runtime/kernel/kernel_includes.h>
+#include <executorch/runtime/platform/assert.h>
+#include "kernels.h"
+
+namespace torch {
+namespace executor {
+namespace native {
+
+#define NNLIB_MAX_DIM 4  /* Add fallback if broadcast and dim > 4 */
+
+Tensor& add_out(
+    RuntimeContext& ctx,
+    const Tensor& a,
+    const Tensor& b,
+    const Scalar& alpha,
+    Tensor& out) {
+  (void)ctx;
+
+  ScalarType a_type = a.scalar_type();
+  ScalarType b_type = b.scalar_type();
+  ScalarType common_type = promoteTypes(a_type, b_type);
+  ScalarType out_type = out.scalar_type();
+
+  ET_CHECK_MSG(a_type == ScalarType::Float, "Input tensor not a float.\n");
+  ET_CHECK_MSG(b_type == ScalarType::Float, "Input tensor not a float.\n");
+  ET_CHECK_MSG(out_type == ScalarType::Float, "Output tensor not a float.\n");
```

Comment:
We might need to fallback instead of erroring for non-float types

Reply:
Done